src/edit/index.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <div class="tce-sprout-video">
+    <element-placeholder
+      v-if="isEmpty"
+      :is-focused="isFocused"
+      :is-disabled="isDisabled"
+      name="Sprout Video component"
+      icon="mdi-video-image"
+      active-placeholder="Use toolbar to upload the video"
+      active-icon="mdi-arrow-up" />
+    <div v-else>
+      <error-message v-if="errorMessage" :message="errorMessage" />
+      <progress-message v-if="!errorMessage && infoMessage" :message="infoMessage" />
+      <div ref="player" class="player d-flex align-center justify-center"></div>
+    </div>
+  </div>
+</template>
+
+<script>
+import createUpload from '../upload';
+import { ELEMENT_STATE } from '../shared';
+import ElementPlaceholder from '../tce-core/ElementPlaceholder.vue';
+import ErrorMessage from './ErrorMessage.vue';
+import get from 'lodash/get';
+import ProgressMessage from './ProgressMessage.vue';
+
+const DEFAULT_ERROR_MSG = 'Something went wrong.';
+const UPLOAD_FAILED_ERROR_MSG = 'Video upload failed. Please try again.';
+const UPLOADING_MSG = 'Video is uploading... Do not leave the page.';
+const PROCESSING_MSG = 'Video is processing...';
+
+export default {
+  name: 'tce-sprout-video',
+  inject: ['$elementBus'],
+  props: {
+    element: { type: Object, required: true },
+    isFocused: { type: Boolean, default: false },
+    isDisabled: { type: Boolean, default: false }
+  },
+  data: () => ({ file: null }),
+  computed: {
+    isEmpty() {
+      const { error, fileName } = this.element.data;
+      return !error && !fileName;
+    },
+    didUploadFail() {
+      const { status } = this.element.data;
+      return status === ELEMENT_STATE.UPLOADING && !this.file;
+    },
+    errorMessage() {
+      const { error } = this.element.data;
+      return this.didUploadFail ? UPLOAD_FAILED_ERROR_MSG : error;
+    },
+    infoMessage() {
+      const { status, playable } = this.element.data;
+      if (status === ELEMENT_STATE.UPLOADING) return UPLOADING_MSG;
+      return playable ? '' : PROCESSING_MSG;
+    },
+    isPreparedToUpload() {
```

Comment:
Sorry for nitpicking. 😅
Is this a tiny bit simpler? 
```suggestion
    isReadyToUpload() {
```